sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1115,49 +1115,94 @@ def get_loopback_capability(self):
+                status_host_output = self.xcvr_eeprom.write(consts.HOST_OUTPUT_LOOPBACK, 0)
+                status_media_input = self.xcvr_eeprom.write(consts.MEDIA_INPUT_LOOPBACK, 0)
+                status_media_output = self.xcvr_eeprom.write(consts.MEDIA_OUTPUT_LOOPBACK, 0)
+                return all([status_host_input, status_host_output, status_media_input, status_media_output])
+
+            if loopback_mode == 'host-side-input-none':
+                return self.xcvr_eeprom.write(consts.HOST_INPUT_LOOPBACK, host_input_val & ~lane_mask)
+
+            if loopback_mode == 'host-side-output-none':
+                return self.xcvr_eeprom.write(consts.HOST_OUTPUT_LOOPBACK, host_output_val & ~lane_mask)
+
+            if loopback_mode == 'media-side-input-none':
+                return self.xcvr_eeprom.write(consts.MEDIA_INPUT_LOOPBACK, media_input_val & ~lane_mask)
+
+            if loopback_mode == 'media-side-output-none':
+                return self.xcvr_eeprom.write(consts.MEDIA_OUTPUT_LOOPBACK, media_output_val & ~lane_mask)
         else:
-            return False
+            if loopback_capability['simultaneous_host_media_loopback_supported'] is False:
+                if any(['host' in loopback_mode and (media_input_val or media_output_val),
+                        'media' in loopback_mode and (host_input_val or host_output_val)]):
+                    txt = 'Simultaneous host media loopback is not supported\n'
+                    txt += f'host_input_val:{host_input_val:02x}, host_output_val:{host_output_val:02x}, '
+                    txt += f'media_input_val:{media_input_val:02x}, media_output_val:{media_output_val:02x}\n'
+                    logger.error(txt)
+                    return False
+
+            if loopback_mode == 'host-side-input' and host_input_support:
+                return self.xcvr_eeprom.write(consts.HOST_INPUT_LOOPBACK, host_input_val | lane_mask)
+
+            if loopback_mode == 'host-side-output' and host_output_support:
+                return self.xcvr_eeprom.write(consts.HOST_OUTPUT_LOOPBACK, host_output_val | lane_mask)
+
+            if loopback_mode == 'media-side-input' and media_input_support:
+                return self.xcvr_eeprom.write(consts.MEDIA_INPUT_LOOPBACK, media_input_val | lane_mask)
+
+            if loopback_mode == 'media-side-output' and media_output_support:
+                return self.xcvr_eeprom.write(consts.MEDIA_OUTPUT_LOOPBACK, media_output_val | lane_mask)
+
+        txt = f'Failed to set {loopback_mode} loopback, lane_mask:{lane_mask:02x}\n'
+        txt += f'host_input_support:{host_input_support}, host_output_support:{host_output_support}, '
+        txt += f'media_input_support:{media_input_support}, media_output_support:{media_output_support}\n'
+        txt += f'host_input_val:{host_input_val:02x}, host_output_val:{host_output_val:02x}, '
+        txt += f'media_input_val:{media_input_val:02x}, media_output_val:{media_output_val:02x}\n'
```

Comment:
@mihirpat1 @xinyulin can we write 4 different functions
`set_host_input_loopback()`
`set_host_ouput_loopback()`
`set_media_output_loopback()`
`set_media_input_loopback()`

And use these functions to make code more readable

Reply:
@prgeor Yes, that would be much clearer to read. Do you think we should keep the `set_loopback_mode() `function since it has been there for years, or should we remove it and replace it with the four functions you proposed?

`set_host_input_loopback(lane_mask, enable)`
`set_host_ouput_loopback(lane_mask, enable)`
`set_media_output_loopback(lane_mask, enable)`
`set_media_input_loopback(lane_mask, enable)`

Reply:
@xinyulin 

```
def set_loopback_mode(self, loopback_mode, lane_mask = 0xff):
    if (loopback_mode == host-side-ouput") :
            set_host_ouput_loopback(lane_mask, enable)
    else if  (loopback_mode == host-side-input") :
            set_host_input_loopback(lane_mask, enable)
   else if (loopback_mode == media-side-input") :
           set_media_output_loopback(lane_mask, enable)
```

Reply:
@prgeor Thank you for the suggestions. It looks more clear and readable now.